### ISSUE TITLE
Show manufacturers "colored" in PinballY

### ISSUE THREAD

**Issue by fR33Styler-1971** [vbousquet/flexdmd#18]
Dear Vincent,

i´ve [created some "manufacurer" animations ](https://www.flippermarkt.de/community/forum/threads/pinbally-eigene-scripte-diskussion.230127/#post-1850075)for your very cool script.

I think about render the stuff in a little higher resulution (896x224) and try a few things... but before i do all this work: do you think it´s possible to use "colored animations" with flexdmd at PinballY?

I (have to) use PinballY like a rom:

instead (part of main.js):
```
function UpdateDMD() {
    if (updater !== undefined) clearTimeout(updater);
    updater = undefined;

    if (dmd == null) {
        dmd = createAutomationObject("FlexDMD.FlexDMD");
        dmd.GameName = "";
        dmd.Width = 128;
        dmd.Height = 32;
        dmd.Show = true;
        dmd.Run = true;
        udmd = dmd.NewUltraDMD();
    }
```

this:
```
function UpdateDMD() {
    if (updater !== undefined) clearTimeout(updater);
    updater = undefined;

    if (dmd == null) {
        dmd = createAutomationObject("FlexDMD.FlexDMD");
        dmd.GameName = "PinballY";
        dmd.Width = 128;
        dmd.Height = 32;
        dmd.Show = true;
        dmd.Run = true;
        udmd = dmd.NewUltraDMD();
    }
```

and in dmddevice.ini:
[PinballY]
virtualdmd left = 358
virtualdmd top = -369
virtualdmd width = 703
virtualdmd height = 337

this brings me to the Idea/Question... how could we handle this as "color rom" in PinballY... 

Any Idea/Direction is welcome

Best regards
Ralf

**Comment by vbousquet** [comment]
The animations are very cool ! I have included them with the script and gave you credit for them (if it's not ok for you, just tell me, and I will rmeove them).

I'm not sure to understand what you want to do.

The ```GameName``` is used to identify the DMD in FlexDMD and moreover in Freezy's DMD to load the right configuration. It used to be broken (changing it would cause a crash to a bug in the DMD lifecycle), but it is now fixed (1.8+) and used as I think it should (i.e. use the DMD corresponding to the selected table, with it's frame, coloring,...).

If you want to use a custom config of the DMD for PinballY, you can replace the code that set ```GameName``` according to the rom and use a fixed value like you propose: it should perfectly work.

If you want to change the resolution of the DMD to use larger images, then this is not the config of Freezy's DMD (which control the ouput, for example, position on the screen, size of dots,...) but the one of FlexDMD (which control the rendering). The line to change are:
```
dmd.Width = 128;
dmd.Height = 32;
```

**Comment by fR33Styler-1971** [comment]
Sure... it is very "ok" -> feel free to use the animations, and i´ll create new ones.. if other manufacturers are needed ;-)

Maybe briefly on the background (I think I worded it a bit unfortunate):
 
What I had described above was only "background information" (so no direct question)... it works perfectly with "PinballY" as dmd.GameName on my setup!

I also need that, because that in PinballY everything lies correctly on top of each other (above the DMD videos) since I have a "portrait" setup in Windows (and thus the position values go into "minus" and not every software can handle it).... but everything looks great... no probleme here (just background information).

My question was rather: if I will create the animations "new and in colour"... can I also display them like this in (also with my setup)?

And if so, would that also work in 896x224 pixel (if the animations had this size)?

So in other words: if flexdmd is able to recognise the colours from the GIF? (which is currently not possible by default... e.g. the "ZEN-Studios" animation has "red" included)....

Because (my thoughts): PinballY is not a "VPinMAME + altcolor (pin2dmd.pal)", I asked myself... how and if that would be feasible.

Looking forward to any ideas/suggestions... 

Greetings and have a nice Sunday
Ralf

**Comment by vbousquet** [comment]
The good news is that yes, if I understand right, everything you would like to do can be easily done with the current release.
To get color (you can alreayd try with the existing videos), simply change the line ```dmd.RenderMode = 1; // 0 = Gray 4 shades, 1 = Gray 16 shades, 2 = Full color``` with the wanted value. To change the output size, simply change the values in ```dmd.Width = 128; dmd.Height = 32;``` and that's all !

So, if you want 896x224 dots, you should have:
```
		dmd.RenderMode = 2; // 0 = Gray 4 shades, 1 = Gray 16 shades, 2 = Full color
		dmd.Width = 896;
		dmd.Height = 224;
```

This is quite a lot of dots; usually you would have less than that, like 224x56 in order for the dots to be visible. It's up to your liking. The main problem is that the size of the text will not change (in number of dots) and therefore will look smaller so a very high resolution may look somewhat to small.

I will close for now, feel free to reopen if needed.

Regards

**Comment by fR33Styler-1971** [comment]
Thx a lot.. it worked and looks great - thank you very much... also the new resolution looks much better in Animation (more details).

I now have two problems thats why i take your suggestion to re-open...

1) dmd crashes when i put a "tablename.png" into "titels-folder"
2) by change the resolution to 896 -> everything else is very small now, i didn´t found where to change the size (should be 7 times more now).... any idea (dmd.Size = false also didn´t helped )?

If i´ll find a solution... i´ll create all animations in a better quality and colored.

Regards

**Comment by fR33Styler-1971** [comment]
Any idea with the 2nd point (by change the resolution to 896 -> everything else is very small now, i didn´t found where to change the size)?

When i got you text (stats etc.) fitting to 896x224 i´ll create new animations ;)
